modules/transport.js

```diff
@@ -179,36 +160,32 @@ __private.receiveSignature = function(query, cb) {
  * @private
  * @implements {library.schema.validate}
  * @implements {__private.receiveTransaction}
- * @param {Object} query - Contains transactions
+ * @param {Array} transactions - List of transactions
  * @param {peer} peer
  * @param {string} extraLogMessage
  * @param {function} cb
  * @return {setImmediateCallback} cb, err
  */
-__private.receiveTransactions = function(query, peer, extraLogMessage, cb) {
-	var transactions;
-
-	transactions = query.transactions;
-
+__private.receiveTransactions = function(
+	transactions,
+	peer,
+	extraLogMessage,
+	cb
+) {
 	async.eachSeries(
 		transactions,
 		(transaction, eachSeriesCb) => {
-			if (!transaction) {
-				return setImmediate(
-					eachSeriesCb,
-					'Unable to process transaction. Transaction is undefined.'
-				);
+			if (transaction) {
+				transaction.bundled = true;
 			}
-			transaction.bundled = true;
-
 			__private.receiveTransaction(transaction, peer, extraLogMessage, err => {
 				if (err) {
 					library.logger.debug(err, transaction);
 				}
-				return setImmediate(eachSeriesCb, err);
+				return setImmediate(eachSeriesCb);
```

Comment:
Are we sure that we are just logging the error here and not returning that error to callback?

Reply:
That was the initial idea but I'm not so sure now. If we do want to pass back errors, we should come up with a format for sending back the list of transactions within the batch/array which failed.

Reply:
I had a discussion with @MaciejBaj about this. For private peer-to-peer batch transactions, the node never sends back error responses to the peer (even if some transactions fail). It's only for propagating transactions throughout the network.